src/evict/evict_file.c

```diff
@@ -87,9 +87,13 @@ __wt_evict_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
         case WT_SYNC_DISCARD:
             /*
              * Discard the page regardless of whether it is dirty.
+             *
+             * If the page has a page deleted structure, we are discarding the page that is cleaned
+             * by a checkpoint.
              */
             WT_ASSERT(session, F_ISSET(dhandle, WT_DHANDLE_DEAD) ||
-                F_ISSET(S2C(session), WT_CONN_CLOSING) || __wt_page_can_evict(session, ref, NULL));
```

Comment:
Instead of removing the check, should we make the check smarter and __wt_page_can_evict(session, ref, NULL)) should return true if the truncate has "restored from ds" flag set ?

Reply:
Unfortunately, __wt_page_can_evict is used in a lot of places and the change there is risky and will change the behaviour in evicition.